arch/posix/CMakeLists.txt

```diff
@@ -0,0 +1,15 @@
+zephyr_cc_option_ifdef(CONFIG_LTO -flto)
+
+zephyr_ld_option_ifdef(CONFIG_COVERAGE
+	-lgcov
+	)
+zephyr_compile_options(
+  -m32
+  -MMD -MP -D_POSIX_C_SOURCE=199309
+  ${TOOLCHAIN_C_FLAGS}
+  ${ARCH_FLAG}
+  -include ${PROJECT_SOURCE_DIR}/arch/posix/include/posix_cheats.h
```

Comment:
This construct should only be used when there is no reasonable way to express this with  #include pragmas.

I don't know the context, but I hope that adding this has not been done lightly.

At the very least, there should be a disclaimer in the file explaining why it was necessary to use -include so that when -include causes issues in the future the guy that needs to deal with that can decide whether getting rid of -include is the simplest solution.

Reply:
The reason to do it this way, is because in this manner it is transparent to the application.
Otherwise posix_cheats.h needs to be included in all the applications' files which define main( ), and in any app file which uses the pthreads like API provided by Zephyr ( include/posix/pthread.h / kernel/pthread.c ) [And any future API added to Zephyr which will clash with the native POSIX API] . It would also need to be included in a few zephyr kernel files.

But I do agree that a comment explaining this would help => will add it.

Reply:
I suspect it is possible to solve the main problem by changing the entry point
and that it is possible to solve the API problem by modifying include paths,
but I am OK with this change as long as it is documented in such a way that
we can fix it later.